### DISCUSSION TITLE
Pre-training weights for YOLOv8 multispectral target detection algorithm

### DISCUSSION THREAD

**Discussion by luckyboby12** [ultralytics/ultralytics#18151] [General]
I have developed a multispectral target detection algorithm based on yolov8, how should I use the pre-training weights during model training?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @luckyboby12, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it.

If this is a custom training ❓ Question, such as yours, please provide as much information as possible to help us assist you better. For example, details about your dataset (including image examples), the pretraining weights you are using (if applicable), any modifications made to the YOLOv8 architecture, and your training logs would be very useful. Additionally, you can verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to optimize your process.

Join the Ultralytics community where it suits you best. For real-time chat about your project, join us on [Discord](https://discord.com/invite/ultralytics) 🎧. For thoughtful, ongoing discussions, visit [Discourse](https://community.ultralytics.com). Or share your insights on our [Subreddit](https://reddit.com/r/Ultralytics)!

## Upgrade

Make sure your `ultralytics` package version is up to date to ensure compatibility and access to the latest features and improvements. You can upgrade using the following command (requires [Python>=3.8](https://www.python.org/) and [PyTorch>=1.8](https://pytorch.org/get-started/locally/)):

```bash
pip install -U ultralytics
```

## Environments

YOLOv8 can be run on various environments to suit your needs, whether you're training locally or using a cloud service:

- **Notebooks** with free GPU access: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to help you get started 😊, but don't worry – an Ultralytics engineer will follow up soon to assist you further if needed!

**Comment by glenn-jocher** [comment]
@luckyboby12 to use pre-trained weights for your YOLOv8-based multispectral target detection algorithm, specify the weights file (e.g., `yolov8n.pt`) when loading the model in your training script or CLI command. For example, in Python: `model = YOLO("yolov8n.pt")`. You can then fine-tune the model on your custom dataset. For more details, refer to the [Train Mode documentation](https://docs.ultralytics.com/modes/train/).

**Reply by luckyboby12** [reply]
My network structure has changed a lot. Can I still use the pre-training weight of yolov8n.pt at this time

**Reply by glenn-jocher** [reply]
@luckyboby12 yes, you can use `yolov8n.pt` as pre-trained weights, but ensure your modified network's layers and dimensions are compatible with the weights to avoid mismatches. You may need to fine-tune or partially load weights for best results.

**Reply by luckyboby12** [reply]
I designed the feature extraction backbone network of yolov8 into a two-stream structure to meet the simultaneous loading of visible and infrared images.

**Reply by luckyboby12** [reply]
Hello ! In fact, I am not sure whether the effect of pre-training weight is effective, and sometimes the effect of using pre-training weight is not necessarily good. I want to know how the pre-training weight works.

**Reply by glenn-jocher** [reply]
Pre-trained weights provide a starting point for training by leveraging learned features from a large dataset, which can accelerate convergence and improve performance, especially with limited data. However, their effectiveness depends on the similarity between the pre-training dataset and your target dataset. Fine-tuning can help adapt these weights to your specific task.
